src/data/cached-history.ts

```diff
@@ -191,6 +197,8 @@ const mergeLine = (
           oldLine.data.push(entity);
         }
       });
+      // Replace the cached line data to force an update
+      oldLine.data = [...oldLine.data];
```

Comment:
Since mutating the array doesn't trigger the update I did this to force it.  Is there a better way?

Reply:
This one is even more complex since its arrays in arrays and it has the same issue of only enumerating the new data